sentry-android-timber/src/test/java/io/sentry/android/timber/SentryTimberTreeTest.kt

```diff
@@ -0,0 +1,185 @@
+            assertEquals(SentryLevel.DEBUG, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures info level event`() {
+        val sut = fixture.getSut(SentryLevel.DEBUG)
+        sut.i(Throwable())
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(SentryLevel.INFO, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures warning level event`() {
+        val sut = fixture.getSut(SentryLevel.DEBUG)
+        sut.w(Throwable())
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(SentryLevel.WARNING, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures error level event`() {
+        val sut = fixture.getSut(SentryLevel.DEBUG)
+        sut.e(Throwable())
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(SentryLevel.ERROR, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures fatal level event`() {
+        val sut = fixture.getSut(SentryLevel.DEBUG)
+        sut.wtf(Throwable())
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(SentryLevel.FATAL, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures unknown as debug level event`() {
+        val sut = fixture.getSut(SentryLevel.DEBUG)
+        sut.log(15, Throwable())
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(SentryLevel.DEBUG, it.level)
+        })
+    }
+
+    @Test
+    fun `Tree captures an event with an exception`() {
+        val sut = fixture.getSut()
+        val throwable = Throwable()
+        sut.e(throwable)
+        verify(fixture.hub).captureEvent(check {
+            assertEquals(throwable, it.getExc())
+        })
+    }
+
+    @Test
+    fun `Tree captures an event without an exception`() {
+        val sut = fixture.getSut()
+        sut.e("message")
+        verify(fixture.hub).captureEvent(check {
+            assertNull(it.getExc())
+        })
+    }
+
+    @Test
+    fun `Tree captures an event with origin tag`() {
+        val sut = fixture.getSut()
+        sut.e("message")
+        verify(fixture.hub).captureEvent(check {
+            assertEquals("SentryTimberIntegration", it.getTag("origin"))
+        })
+    }
+
+    @Test
+    fun `Tree captures an event with TimberTag tag`() {
+        val sut = fixture.getSut()
+        Timber.plant(sut)
+        // only available thru static class
+        Timber.tag("tag")
+        Timber.e("message")
+        verify(fixture.hub).captureEvent(check {
+            assertEquals("tag", it.getTag("TimberTag"))
+        })
+    }
+
+    @Test
+    fun `Tree captures an event without TimberTag tag`() {
+        val sut = fixture.getSut()
+        Timber.plant(sut)
+        Timber.e("message")
+        verify(fixture.hub).captureEvent(check {
+            assertNull(it.getTag("TimberTag"))
+        })
+    }
+
+    @Test
+    fun `Tree captures an event with given message`() {
+        val sut = fixture.getSut()
+        sut.e("message")
```

Comment:
`m`: I would also add a test with a message containing `\n`

Reply:
the test would pretty much fail because of https://github.com/getsentry/sentry-android/pull/464#discussion_r443786703

so we'd need to talk if we want to go with that approach or not.

Reply:
I mean, it'd log only the 1st line

Reply:
no need for that now as we just use the raw given message by timber